ui/src/app/utils/index.tsx

```diff
@@ -533,6 +536,19 @@ export function validateInputForMySQL(searchString: string): Array<string> {
     // unclosed quote
     inputErrors.add('There is an unclosed " in the search string');
   }
+  // check each endsWith term - terms that start with *
+  searchString.split(' ').forEach((word) => {
+    if (word.substring(1).length < searchTrigger && word.startsWith('*')) {
+      inputErrors.add(
+        `Minimum criteria search length for ends with term is two characters after '*' [${word}]`
+      );
+    }
+    if (word.match(/[+*-]{2,}|^\*.*[+*-]$/)) {
+      inputErrors.add(
+        `Search term cannot contain consecutive special characters [${word}]`
```

Comment:
This will show an error for non-consecutive special characters in some cases.
![Screen Shot 2022-09-13 at 10 30 20 PM](https://user-images.githubusercontent.com/40036095/190053292-5b6009bb-9eed-4a93-83a0-07a97cb10eba.png)

Reply:
yes you cannot have *word* as it does not make sense that this is a endsWith term, since the * is replaced by % so %word% -> is 'contains' and not endsWith. The backend drops these terms, if present. So added the same validation rule to front end. If we allow *diabete* -> backend drops the term since %diabete% is not endsWith, but 'contains'. This way it is consistent.

Reply:
maybe we should give a different error message for this specific case.  done